src/EnergyPlus/Plant/DataPlant.hh

```diff
@@ -333,10 +331,8 @@ struct DataPlantData : BaseGlobalStruct
         this->PlantManageHalfLoopCalls = 0;
         this->PlantLoop.deallocate();
         this->PlantAvailMgr.deallocate();
-        this->VentRepPlantSupplySide.deallocate();
-        this->VentRepPlantDemandSide.deallocate();
-        this->VentRepCondSupplySide.deallocate();
-        this->VentRepCondDemandSide.deallocate();
+        std::for_each(this->VentRepPlant.begin(), this->VentRepPlant.end(), [](Array1D<DataPlant::ReportLoopData> &Array1DObject) { Array1DObject.deallocate();});
+        std::for_each(this->VentRepCond.begin(), this->VentRepCond.end(), [](Array1D<DataPlant::ReportLoopData> &Array1DObject) { Array1DObject.deallocate();});
```

Comment:
Don't know if this was the best way to `deallocate()` the `Array1D` objects in the `std::array`...

Reply:
`this->VentRepPlant[static_cast<int>LoopSideLocation::Demand].deallocate();` etc. is also good. A `for_each` with a `lambda` is a bit of a sledgehammer for only two elements.

Reply:
Thanks, I will change it!